spec/assently/client/get_case_spec.rb

```diff
@@ -0,0 +1,84 @@
+require "luhn"
+require "securerandom"
+require "spec_helper"
+
+require "assently/client"
+require "assently/case"
+require "assently/document"
+require "assently/party"
+
+module Assently
+  describe Assently::Client do
+    let :client do
+      Assently::Client.new ENV["ASSENTLY_API_KEY"], ENV["ASSENTLY_API_SECRET"], :test
+    end
+
+    describe "#get_case" do
+      describe "with a case_id as argument" do
+        it "sends the getcase query with the case id json" do
+          allow(client).to receive :get
+
+          case_id = SecureRandom.uuid
+
+          client.get_case case_id
+
+          expect(client).to have_received(:get).with "/api/v2/getcase", id: case_id
+        end
+      end
+
+      describe "when the case exists", vcr: { cassette_name: "Assently_Client/_get_case/case_exists" } do
+        let (:case_id) { "my-case-id" }
+
+        before do
+          create_case case_id
+        end
+
+        describe "result" do
+          it "is successful" do
+            result = client.get_case case_id
+
+            expect(result.success?).to be true
+          end
+
+          it "has the url" do
+            result = client.get_case case_id
+
+            expect(result).to be_a Assently::Client::SuccessResult
+            expect(result.response["Parties"].first["PartyUrl"]).to match(/\Ahttps:\/\/test.assently.com\/c\/.*\z/)
```

Comment:
I think we should return objects that have a known interface instead of the json from the api request.
Reasons: testability and api discoverability.